src/test/java/net/snowflake/client/jdbc/ResultSetLatestIT.java

```diff
@@ -132,74 +132,79 @@ public void testMemoryClearingAfterInterrupt() throws Throwable {
-      ResultSet resultSet =
-          stmt.executeQuery(
-              "select randstr(100, random()) from table(generator(rowcount => " + rowCount + "))");
-      rsList.add(resultSet);
-    }
-    // Assert that all resultSets exist and can successfully download the needed chunks without
-    // hanging
-    for (int i = 0; i < stmtCount; i++) {
-      rsList.get(i).next();
-      assertTrue(Pattern.matches("[a-zA-Z0-9]{100}", rsList.get(i).getString(1)));
-      rsList.get(i).close();
+      for (int i = 0; i < stmtCount; ++i) {
+        ResultSet resultSet =
+            stmt.executeQuery(
+                "select randstr(100, random()) from table(generator(rowcount => "
+                    + rowCount
+                    + "))");
+        rsList.add(resultSet);
+      }
+      // Assert that all resultSets exist and can successfully download the needed chunks without
+      // hanging
+      for (int i = 0; i < stmtCount; i++) {
+        rsList.get(i).next();
+        assertTrue(Pattern.matches("[a-zA-Z0-9]{100}", rsList.get(i).getString(1)));
+        rsList.get(i).close();
+      }
+      // set memory limit back to default invalid value so it does not get used
+      connection
+          .unwrap(SnowflakeConnectionV1.class)
+          .getSFBaseSession()
+          .setMemoryLimitForTesting(SFBaseSession.MEMORY_LIMIT_UNSET);
     }
-    // set memory limit back to default invalid value so it does not get used
-    connection
-        .unwrap(SnowflakeConnectionV1.class)
-        .getSFBaseSession()
-        .setMemoryLimitForTesting(SFBaseSession.MEMORY_LIMIT_UNSET);
-    connection.close();
   }
 
   /** This tests that the SnowflakeChunkDownloader doesn't hang when memory limits are low. */
   @Test
   public void testChunkDownloaderSetRetry() throws SQLException {
     int stmtCount = 3;
     int rowCount = 170000;
-    Connection connection = getConnection();
-    connection
-        .unwrap(SnowflakeConnectionV1.class)
-        .getSFBaseSession()
-        .setMemoryLimitForTesting(1 * 1024 * 1024);
-    connection
-        .unwrap(SnowflakeConnectionV1.class)
-        .getSFBaseSession()
-        .setOtherParameter(SessionUtil.JDBC_CHUNK_DOWNLOADER_MAX_RETRY, 1);
-    // Set memory limit to low number
-    // open multiple statements concurrently to overwhelm current memory allocation
-    for (int i = 0; i < stmtCount; ++i) {
+    try (Connection connection = getConnection()) {
+      connection
+          .unwrap(SnowflakeConnectionV1.class)
+          .getSFBaseSession()
+          .setMemoryLimitForTesting(1 * 1024 * 1024);
+      connection
+          .unwrap(SnowflakeConnectionV1.class)
+          .getSFBaseSession()
+          .setOtherParameter(SessionUtil.JDBC_CHUNK_DOWNLOADER_MAX_RETRY, 1);
+      // Set memory limit to low number
+      // open multiple statements concurrently to overwhelm current memory allocation
       Statement stmt = connection.createStatement();
```

Comment:
This should be moved to the try below with the resultset inside the for-loop. It looks like the point here is to make multiple statements whereas this would just execute the same statement multiple times.

Reply:
Fixed.